pandas/tests/extension/test_categorical.py

```diff
@@ -117,8 +117,11 @@ class TestConstructors(base.BaseConstructorsTests):
 
 
 class TestReshaping(base.BaseReshapingTests):
-    def test_concat_with_reindex(self, data):
-        pytest.xfail(reason="Deliberately upcast to object?")
+    def test_concat_with_reindex(self, request, data):
+        request.node.add_marker(
+            pytest.mark.xfail(reason="Deliberately upcast to object?")
+        )
+        assert False
```

Comment:
shouldn't this should be the super call. Also why the request.node.add_marker pattern, does the pytest.mark.xfail decorator not work in this case?

Reply:
Thanks, decorator should be preferred here, will update. Not sure what you mean by "the super call".

Reply:
we still want to run the test for the xfail to be strict

```suggestion
        super().test_concat_with_reindex(data)
```

Reply:
Ah, I see. master currently doesn't run the test either, but I've verified it does fail and the message is appropriate. Seems okay to add that in here.

Reply:
> master currently doesn't run the test either

it may also be worth changing the skips in this test file to xfails as well, separate PR OK.

Reply:
Opened #38904 to track.

Reply:
thanks.